File: 20-valid-parentheses/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        
        stack<char> st;

        st.push(s[0]);

        for(int i = 1; i < s.length() ; i++){

            if(s[i] == ')' || s[i] == ']' || s[i] == '}'){
                if(st.empty())return false;
                
                char tel = st.top();
                st.pop();
                if((s[i] == ')' && tel != '(') || (s[i] == '}' && tel != '{') || (s[i] == ']' && tel != '[')){
                    return false;
                }
            
            }
            else{
                st.push(s[i]);
            }
            

        }

        return (st.size() > 0)?false:true;
    }
};

File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        helper = {")":"(","}":"{","]":"["}
        for c in s:
            if c in helper:
                if stack and stack[-1] == helper.get(c,False):
                    stack.pop()
                else :
                    return False
            else:
                stack.append(c)
        return len(stack) == 0